commit 8d1af215e11a3a4f932a5a4d162293601311dd6d (refs/changes/16/3316/2)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-02-09T14:33:21+00:00 (1 year, 8 months ago)
    
    OS-6552 devnet_filldir should only call sdev_cleandir on directories, again

diff --git a/usr/src/uts/common/fs/dev/sdev_netops.c b/usr/src/uts/common/fs/dev/sdev_netops.c
index 41441ec52d..a0edd668f2 100644
--- a/usr/src/uts/common/fs/dev/sdev_netops.c
+++ b/usr/src/uts/common/fs/dev/sdev_netops.c
@@ -21,7 +21,7 @@
 /*
  * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
- * Copyright (c) 2014, Joyent, Inc.  All rights reserved.
+ * Copyright (c) 2018, Joyent, Inc.  All rights reserved.
  */
 
 /*
@@ -487,7 +487,7 @@ devnet_filldir(struct sdev_node *ddv)
 		/*
 		 * Clean out everything underneath before we remove ourselves.
 		 */
-		if (SDEVTOV(ddv)->v_type == VDIR) {
+		if (SDEVTOV(dv)->v_type == VDIR) {
 			ret = sdev_cleandir(dv, NULL, 0);
 			ASSERT(ret == 0);
 		}
